test/smartseq2_single_sample/pr/ValidateSmartSeq2SingleCell.wdl

```diff
@@ -0,0 +1,43 @@
+
+task ValidateSmartSeq2SingleCell {
+      File counts
+      String expected_counts_hash
+
+      File alignment_summary_metrics
+      String expected_alignment_summary_metrics_hash
+
+  command <<<
+
+    # catch intermittent failures
+    set -eo pipefail
+
+    # calculate hashes; awk is used to extract the hash from the md5sum output that contains both
+    # a hash and the filename that was passed. We parse the first 7 columns because a bug in RSEM
+    # makes later columns non-deterministic.
+    counts_hash=$(cut -f 1-7 "${counts}" | md5sum | awk '{print $1}')
+
+    # the picard metrics are contained in a .tar.gz file; in addition to the above processing,
+    # this unzips the archive, and parses it with awk to remove all the run-specific comment lines (#)
+    picard_metrics_hash=$(cat "${alignment_summary_metrics}" | awk 'NF && $1!~/^#/' | md5sum | awk '{print $1}')
+
+    if [ "$counts_hash" != "${expected_counts_hash}" ]; then
+      >&2 echo "counts_hash ($counts_hash) did not match expected hash (${expected_counts_hash})"
```

Comment:
I'm going back and forth as to whether I would want to see the diff here or not along with the mismatching hashes just to provide a little more debugging information or do we expect someone to dig into the files themselves if their test fails?

Reply:
Well, one concern is that these files could sometimes be binary, in what case a diff would overload the log. What about the head of a diff? 

I do imagine that people would often need to dig into their files, but wouldn't be opposed to making life easier in cases where a diff solves the problem.